src/lib.rs

```diff
@@ -65,6 +67,22 @@ pub trait Body {
     fn size_hint(&self) -> SizeHint {
         SizeHint::default()
     }
+
+    /// Returns future that resolves to next data chunk, if any.
+    fn next_data(&mut self) -> NextData<'_, Self>
```

Comment:
I'd like to see this renamed to maybe `next` and `next_trailers` to `trailers`.

Reply:
done